--- a/klippy/extras/heater_fan.py
+++ b/klippy/extras/heater_fan.py
@@ -21,6 +21,10 @@
         self.fan_speed = config.getfloat("fan_speed", 1., minval=0., maxval=1.)
         self.min_speed = config.getfloat("min_speed", 0., minval=0., maxval=1.)
         self.probe_speed = config.getfloat("probe_speed", self.fan_speed, minval=0., maxval=1.)
+        self.target_temp_threshold = config.getfloat(
+            'target_temp_threshold', 9999., minval=0.)
+        self.target_temp_speed = config.getfloat(
+            'target_temp_speed', 0., minval=0., maxval=1.)
         self.temp_speed_table = config.getlists('temp_speed_table', None, seps=(',', '\n'), count=5, parser=float)
         self.original_fan_speed = None
         self.last_speed = 0.
@@ -143,6 +147,12 @@
                 if matching_rule is not None:
                     speed = max(self.min_speed, min(
                         self.temp_speed_table[matching_rule][4], 1.0))
+                elif self.target_temp_speed > 0:
+                    for heater in self.heaters:
+                        _, target_temp = heater.get_temp(eventtime)
+                        if target_temp >= self.target_temp_threshold:
+                            speed = self.target_temp_speed
+                            break
             else:
                 for rule in self.temp_speed_table:
                     temp_threshold, target_temp_threshold, satisfied_heater_threshold, heater_count_threshold, rule_speed = rule
